docs/protocol_specification/tangle.md

```diff
@@ -0,0 +1,372 @@
+To overcome this limitation, we propose the `approval switch`. The idea is to minimize honest messages along with transactions getting orphaned, by assigning a different meaning to the parents of a message.
+
+### Detailed design
+Each message can express two levels of approval with respect to its parents:
+* **Strong**: it defines approval for both the referenced message along with its entire past cone.
+* **Weak**: it defines approval for the referenced message but not for its past cone.
+
+Let's consider the following example:
+
+![](https://i.imgur.com/rfpnkcg.png)
+
+Message *D* contains a transaction that has been rejected, thus, due to the monotonicity rule, its future cone must be orphaned. Both messages *F* (transaction) and *E* (data) directly reference *D* and, traditionally, they should not be considered for tip selection. However, by introducing the approval switch, these messages can be picked up via a **weak** reference as messages *G* and *H* show.
+
+We define two categories of eligible messages:
+- **Strong message**:
+    * It is eligible
+    * Its payload is liked with level of knowledge >=2
+    * Its branch is **liked** with level of knowledge >= 2
+- **Weak message**:
+    * It is eligible
+    * Its payload is liked with level of knowledge >=2
+    * Its branch is **not liked** with level of knowledge >= 2
+
+We call *strong approver of x* (or *strong child of x*) any strong message *y* approving *x* via a strong reference. Similarly, we call *weak approver of x* (or *weak child of x*) any strong message *y* approving *x* via a weak reference.
+
+### TSA
+We define two separate tip types:
+* **Strong tip**:
+    * It is a strong message
+    * It is not directly referenced by any strong message via strong parent
+* **Weak tip**:
+    * It is a weak message
+    * It is not directly referenced by any strong message via weak parent
+
+Consequently, a node keeps track of the tips by using two distinct tips sets:
+* **Strong tips set**: contains the strong tips
+* **Weak tips set**: contains the weak tips
+
+Tips of both sets must be managed according to the local perception of the node. Hence, a strong tip loses its tip status if it gets referenced (via strong parent) by a strong message. Similarly, a weak tip loses its tip status if it gets referenced (via weak parent) by a strong message. This means that weak messages approving via either strong or weak parents, do not have an impact on the tip status of the messages they reference.
+
+### Branch management
```

Comment:
Should we keep this section? There's a Finality section below and referencing the consensus spec containing stuff here.

Reply:
Yes but this is very specific of how we inherit the weight with the approval switch. I don't think it's explained anywhere else how this exactly works. So I'd just keep it here for now.